### PR TITLE
Make python optional via flags

### PR DESCRIPTION
**Pull request by sammorris81**
Add a flag to determine if we should install the Python interface, and add additional python3-dev dependency to the install instructions.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -88,8 +94,13 @@ endif()
 # something like:
 # OPENSSL_ROOT_DIR=/usr/local/opt/openssl cmake ..
 include_directories(
-    ${GFLAGS_INCLUDE_DIRS} ${GLOG_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIR}
+    ${GFLAGS_INCLUDE_DIRS} ${GLOG_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIR})
+
+if (WITH_PYTHON)
+  include_directories(
```
Comment (jmr) [inline CMakeLists.txt:100]: put this on one line since it fits now

File: README.md [hunk 2]
```diff
@@ -133,11 +133,12 @@ For more info read: [The CMake Cache](https://cmake.org/cmake/help/latest/guide/
 If you want the Python interface, you will also need:
```
Comment (jmr) [inline README.md:133]: you'll also need to run cmake with -DWITH_PYTHON

**Review by jmr — CHANGES_REQUESTED** [review]

**Comment by sammorris81** [comment]
All suggestions addressed. Please let me know if there's anything additional.

**Comment by jmr** [comment]
Thanks for the improvement/fix!

**Comment by jmr** [comment]
Sorry, one more thing I just thought of. Can you add python back here, so it's still tested, since it now defaults to off??

https://github.com/google/s2geometry/blob/master/.travis.yml#L162

**Comment by sammorris81** [comment]
Added -DWITH_PYTHON=ON to the test. Please let me know if there's anything additional you would like.

**Pull request merged**